### PR TITLE
Adding IDEA_o1_v02 and its vertex detector

### PR DESCRIPTION
**Pull request by armin-ilg**
BEGINRELEASENOTES
- Adding the IDEA_o1_v02 detector in key4hep, for the start just the vertex detector (vertex barrel + outer barrel) and the machine components, some place holder detectors otherwise.

ENDRELEASENOTES

To do's still:
- Remove overlap between vertex support (imported from CAD) and vertex barrel
- Cross check again all vertex dimensions
- More detailed implementation of barrel (individual ARCADIA chips in vertex barrel, individual ATLASPix3 in outer barrel)
- Add README.md

Longer term:
- Add vertex digitisation (Armin)
- Add silicon wrapper with timing information (Armin)
- Add DRC, drift chamber, muon chambers

### CODE REVIEW

**Comment by andresailer** [comment]
* What are the se "*.swp" binary files?
* Could you flatten the history please? I.e.: No `merge from lcgeo/master` commits, please use rebase.

**Review by BrieucF — COMMENTED** [review]
Hi @armin-ilg ! Thanks! Could you 
- move everything from `FCCee/compact/FCCee_IDEA_o1_v01/` to `FCCee/compact/IDEA/IDEA_o1_v01`
- Add a `README.md` under `detector/tracker` explaining briefly what the new detector builders do together with a link to a recent reference (a talk is ok) describing the detector we are trying to model? You can just add two "entries" in the README to describe the two new builders, more will come to describe the other ones.

File: detector/tracker/VertexBarrel_IDEA_o1_v01_geo.cpp
Comment (BrieucF) [inline detector/tracker/VertexBarrel_IDEA_o1_v01_geo.cpp:1]: Is there another way to identify this vertex detector than by using "IDEA" in the name? Ideally the C++ detector builder should not be detector specific...

File: detector/tracker/ZPlanarTracker_geo.cpp
[the remark this answers is not in the document]
Comment (armin-ilg) [reply detector/tracker/ZPlanarTracker_geo.cpp:1]: Thanks for the comments, still need to write the readme, the rest is addressed

File: detector/tracker/VertexBarrel_IDEA_o1_v01_geo.cpp
Comment (armin-ilg) [reply detector/tracker/VertexBarrel_IDEA_o1_v01_geo.cpp:1]: Changed that, let me know if you have a better name in mind

Comment (BrieucF) [reply detector/tracker/VertexBarrel_IDEA_o1_v01_geo.cpp:1]: Thanks Armin! Just a suggestion: 'complex' --> 'detailed'?
If you don't need to modify `ZPlanarTracker_geo.cpp`, can you remove it from the PR?

File: detector/other/SingleShape_geo.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+// All rights reserved.
+//
+// For the licensing terms see $DD4hepINSTALL/LICENSE.
+// For the list of contributors see $DD4hepINSTALL/doc/CREDITS.
+//
+// Author     : M.Frank
+//
+//==========================================================================
+
+// Edited by A. Ilg, to have only the CAD import left and nothing else
```
Comment (BrieucF) [inline detector/other/SingleShape_geo.cpp:14]: Can you add one more comment line here explaining what this code does?

**Comment by BrieucF** [comment]
Can you add a test for the newly introduced detectors, as in https://github.com/key4hep/k4geo/blob/master/lcgeoTests/CMakeLists.txt#L67 ?

File: FCCee/compact/IDEA/IDEA_o1_v01/Beampipe_o1_v01.xml [hunk 2]
```diff
@@ -0,0 +1,150 @@
+<lccdd>
+
+  <info name="FCCee"
+        title="FCCee Beam pipe: taken corresponding to CLD: Beampipe_o4_v04_noNotch_W_n02.xml"
+        author="from ILCSOFT/lcgeo/FCCee/compact/FCCee_o1_v04"
+        url="no"
+        status="development"
+        version="1.0">
+    <comment>A beampipe for FCCee, R(central) = 1.5 cm</comment>
```
Comment (andresailer) [inline FCCee/compact/IDEA/IDEA_o1_v01/Beampipe_o1_v01.xml:9]: Is this true? the R = 1.5 cm?

File: detector/tracker/ZPlanarTracker_geo.cpp
Comment (andresailer) [inline detector/tracker/ZPlanarTracker_geo.cpp:1]: You were not supposed to delete this file, you were supposed to remove your changes from the file.

File: FCCee/compact/IDEA/IDEA_o1_v01/HOMAbsorber_o1_v01.xml
Comment (andresailer) [inline FCCee/compact/IDEA/IDEA_o1_v01/HOMAbsorber_o1_v01.xml:1]: As I recently realised, the HOM Absorber is no longer needed with the low impedance beampipe

**Comment by armin-ilg** [comment]
> * What are these "*.swp" binary files?
> * Could you flatten the history please? I.e.: No `merge from lcgeo/master` commits, please use rebase.

.swp files are deleted. 
I created a flattened branch (https://github.com/armin-ilg/lcgeo/tree/flattened), I can use that one for the merge request if you want.

**Comment by jmcarcell** [comment]
It's better to continue here where there is already discussion. The number of commits doesn't matter, they will all get squashed into a single one at the end

**Comment by armin-ilg** [comment]
> Can you add a test for the newly introduced detectors, as in https://github.com/key4hep/k4geo/blob/master/lcgeoTests/CMakeLists.txt#L67 ?

Done

**Comment by armin-ilg** [comment]
> Hi @armin-ilg ! Thanks! Could you
> 
> * move everything from `FCCee/compact/FCCee_IDEA_o1_v01/` to `FCCee/compact/IDEA/IDEA_o1_v01`
> * Add a `README.md` under `detector/tracker` explaining briefly what the new detector builders do together with a link to a recent reference (a talk is ok) describing the detector we are trying to model? You can just add two "entries" in the README to describe the two new builders, more will come to describe the other ones.

Done, is the readme enough?

File: detector/other/SingleShape_geo.cpp (hunk 1 above)
Comment (armin-ilg) [reply detector/other/SingleShape_geo.cpp]: Removed file

**Comment by armin-ilg** [comment]
@BrieucF @andresailer : I addressed all of the above comments and the test that I added for IDEA works (I replaced the vertex inner barrel support that I had in STL format by .OBJ thus getting rid of the G4Exceptions in the test).

About the beampipe: Shall I take it out for the moment and we add André's when it's ready?
Is there something else needed?

**Comment by BrieucF** [comment]
> @BrieucF @andresailer : I addressed all of the above comments and the test that I added for IDEA works (I replaced the vertex inner barrel support that I had in STL format by .OBJ thus getting rid of the G4Exceptions in the test).
> 
> About the beampipe: Shall I take it out for the moment and we add André's when it's ready? Is there something else needed?

Hi Armin, great, thanks! For the beampipe, I would propose to leave it like this for now and update when we have the final version. 

Could you just re-organize a bit what is inside `FCCee/IDEA/scripts/utils`? There are duplicated and very generic things there which could go to the main [utils](https://github.com/key4hep/k4geo/tree/master/utils) repository. At least `dd4hep2root` should go there and `overlap.mac` should be taken from there (and removed from this PR). In my opinion, whatever is not IDEA specific could also be placed in the main `utils` folder (in some cases one would need to pass the compact file as an argument) but I let you judge. Another possible solution for things which are FCC specific is to have a folder `k4geo/FCCee/scripts/`. I would also propose to have `scripts` and `utils` at the same level (not utils inside scripts).

**Comment by armin-ilg** [comment]
> > @BrieucF @andresailer : I addressed all of the above comments and the test that I added for IDEA works (I replaced the vertex inner barrel support that I had in STL format by .OBJ thus getting rid of the G4Exceptions in the test).
> > About the beampipe: Shall I take it out for the moment and we add André's when it's ready? Is there something else needed?
> 
> Hi Armin, great, thanks! For the beampipe, I would propose to leave it like this for now and update when we have the final version.
> 
> Could you just re-organize a bit what is inside `FCCee/IDEA/scripts/utils`? There are duplicated and very generic things there which could go to the main [utils](https://github.com/key4hep/k4geo/tree/master/utils) repository. At least `dd4hep2root` should go there and `overlap.mac` should be taken from there (and removed from this PR). In my opinion, whatever is not IDEA specific could also be placed in the main `utils` folder (in some cases one would need to pass the compact file as an argument) but I let you judge. Another possible solution for things which are FCC specific is to have a folder `k4geo/FCCee/scripts/`. I would also propose to have `scripts` and `utils` at the same level (not utils inside scripts).

Hi @BrieucF 

I put all utils (including the material budget stuff) into `k4geo/utils` now. Since I saw that since recently there is a [k4geo/scripts](https://github.com/key4hep/k4geo/tree/master/scripts) folder, I put my scripts there as they can be used for all detectors, I hope that's fine. 

To make the scripts easy to use, it would be great to have an environmental variable to point to the k4geo directory, as it's also the case in DD4hep with [DD4hep_DIR](https://github.com/AIDASoft/DD4hep/blob/54dcf46368b664a53b0cb0e7d392b586c81a765d/cmake/thisdd4hep.sh#L74). In this way the scripts can be used from any directory. Could we have that @jmcarcell @tmadlener @andresailer ? (I don't know where to correctly add this). It would be something like
`export k4geo_DIR=$(pwd)
`
in the top k4geo directory [here](https://github.com/key4hep/k4geo/blob/7914eb0eb6dd7195fb4dc7601ed3be720689ae94/scripts/install.sh#L18).

**Comment by BrieucF** [comment]
> > > @BrieucF @andresailer : I addressed all of the above comments and the test that I added for IDEA works (I replaced the vertex inner barrel support that I had in STL format by .OBJ thus getting rid of the G4Exceptions in the test).
> > > About the beampipe: Shall I take it out for the moment and we add André's when it's ready? Is there something else needed?
> > 
> > 
> > Hi Armin, great, thanks! For the beampipe, I would propose to leave it like this for now and update when we have the final version.
> > Could you just re-organize a bit what is inside `FCCee/IDEA/scripts/utils`? There are duplicated and very generic things there which could go to the main [utils](https://github.com/key4hep/k4geo/tree/master/utils) repository. At least `dd4hep2root` should go there and `overlap.mac` should be taken from there (and removed from this PR). In my opinion, whatever is not IDEA specific could also be placed in the main `utils` folder (in some cases one would need to pass the compact file as an argument) but I let you judge. Another possible solution for things which are FCC specific is to have a folder `k4geo/FCCee/scripts/`. I would also propose to have `scripts` and `utils` at the same level (not utils inside scripts).
> 
> Hi @BrieucF
> 
> I put all utils (including the material budget stuff) into `k4geo/utils` now. Since I saw that since recently there is a [k4geo/scripts](https://github.com/key4hep/k4geo/tree/master/scripts) folder, I put my scripts there as they can be used for all detectors, I hope that's fine.
> 
> To make the scripts easy to use, it would be great to have an environmental variable to point to the k4geo directory, as it's also the case in DD4hep with [DD4hep_DIR](https://github.com/AIDASoft/DD4hep/blob/54dcf46368b664a53b0cb0e7d392b586c81a765d/cmake/thisdd4hep.sh#L74). In this way the scripts can be used from any directory. Could we have that @jmcarcell @tmadlener @andresailer ? (I don't know where to correctly add this). It would be something like `export k4geo_DIR=$(pwd) ` in the top k4geo directory [here](https://github.com/key4hep/k4geo/blob/7914eb0eb6dd7195fb4dc7601ed3be720689ae94/scripts/install.sh#L18).

Hi Armin, thanks a lot! For now $K4GEO is only defined in the nightlies but for the current stable stack you can use $LCGEO.

Regarding your last commit, I think there are two left over files: `FCCee/IDEA/scripts/utils/dd4hep2root` and `FCCee/IDEA/scripts/utils/overlap.mac`.

**Comment by armin-ilg** [comment]
> > > > @BrieucF @andresailer : I addressed all of the above comments and the test that I added for IDEA works (I replaced the vertex inner barrel support that I had in STL format by .OBJ thus getting rid of the G4Exceptions in the test).
> > > > About the beampipe: Shall I take it out for the moment and we add André's when it's ready? Is there something else needed?
> > > 
> > > 
> > > Hi Armin, great, thanks! For the beampipe, I would propose to leave it like this for now and update when we have the final version.
> > > Could you just re-organize a bit what is inside `FCCee/IDEA/scripts/utils`? There are duplicated and very generic things there which could go to the main [utils](https://github.com/key4hep/k4geo/tree/master/utils) repository. At least `dd4hep2root` should go there and `overlap.mac` should be taken from there (and removed from this PR). In my opinion, whatever is not IDEA specific could also be placed in the main `utils` folder (in some cases one would need to pass the compact file as an argument) but I let you judge. Another possible solution for things which are FCC specific is to have a folder `k4geo/FCCee/scripts/`. I would also propose to have `scripts` and `utils` at the same level (not utils inside scripts).
> > 
> > 
> > Hi @BrieucF
> > I put all utils (including the material budget stuff) into `k4geo/utils` now. Since I saw that since recently there is a [k4geo/scripts](https://github.com/key4hep/k4geo/tree/master/scripts) folder, I put my scripts there as they can be used for all detectors, I hope that's fine.
> > To make the scripts easy to use, it would be great to have an environmental variable to point to the k4geo directory, as it's also the case in DD4hep with [DD4hep_DIR](https://github.com/AIDASoft/DD4hep/blob/54dcf46368b664a53b0cb0e7d392b586c81a765d/cmake/thisdd4hep.sh#L74). In this way the scripts can be used from any directory. Could we have that @jmcarcell @tmadlener @andresailer ? (I don't know where to correctly add this). It would be something like `export k4geo_DIR=$(pwd) ` in the top k4geo directory [here](https://github.com/key4hep/k4geo/blob/7914eb0eb6dd7195fb4dc7601ed3be720689ae94/scripts/install.sh#L18).
> 
> Hi Armin, thanks a lot! For now $K4GEO is only defined in the nightlies but for the current stable stack you can use $LCGEO.
> 
> Regarding your last commit, I think there are two left over files: `FCCee/IDEA/scripts/utils/dd4hep2root` and `FCCee/IDEA/scripts/utils/overlap.mac`.

Ok, so I found a different way to have the correct paths (`script_folder="$( cd -- "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"`). For the DDCAD imported file I used $K4GEO now, which will work once the PR is done and used in the key4hep nightly. 

And I've now deleted the files you mentioned, sorry!

**Review by BrieucF — APPROVED** [review]
Thanks!

**Comment by armin-ilg** [comment]
Anything else to be changed @andresailer @BrieucF ? 
Would be nice to have IDEA inside k4geo before the summer break.

Cheers,
Armin

File: scripts/install.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
+### Source script to install k4geo 
+
+# using the nightly release
+source /cvmfs/sw-nightlies.hsf.org/key4hep/setup.sh                                                                                                                                                                                                                                                                                                              
+
+script_folder="$( cd -- "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )" # Get the location of this script, to execute correctly other commands
+cd $script_folder/..
+
+echo "Running on xml file $xml"
```
Comment (andresailer) [inline scripts/install.sh:9]:
```suggestion
```

File: scripts/install.sh [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+mkdir build
+cd build
+source $DD4hep_DIR/bin/thisdd4hep.sh
```
Comment (andresailer) [inline scripts/install.sh:12]: This shouldn't be necessary, if this is sourced from the stack.
Is this for compiling against your own DD4hep?

I think this script should be dropped.

File: scripts/simulate.sh [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+if [[ -n "$xml" ]]; then # test to see if not empty
+    ddsim --compactFile ${xml} \
+      --enableGun \
+      --gun.distribution uniform \
+      --gun.energy "10*GeV" \
+      --gun.particle mu- \
+      --numberOfEvents 500 \
+      --outputFile Step1_edm4hep.root
+else
+    echo "argument error, please provide an xml file as input argument!"
```
Comment (andresailer) [inline scripts/simulate.sh:15]:
```suggestion
    echo "argument error, please provide an xml file as input argument!"
    exit 1
```

File: utils/dd4hep2root [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+def main():
+    parser = argparse.ArgumentParser(description='Convert detector')
+    parser.add_argument('-c', '--compact', help='Compact file location(s)',
+                        required=True, type=str, nargs='+')
+    parser.add_argument('-o', '--out', help='Converted file path',
+                        default='detector.root', type=str)
+    args = vars(parser.parse_args())
+
+    convert(args['compact'], args['out'])
```
Comment (andresailer) [inline utils/dd4hep2root:20]:
```suggestion
    args = parser.parse_args()

    convert(args.compact, args.out)
```

That's how this should work, isn't it?

Comment (armin-ilg) [reply utils/dd4hep2root]: Yep, it also worked before though, could propagate it to https://hep-fcc.github.io/fcc-tutorials/master/full-detector-simulations/Visualization/Visualization.html#detector-geometry as well

File: scripts/simulate.sh (hunk 5 above)
Comment (armin-ilg) [reply scripts/simulate.sh:15]: Done

File: scripts/install.sh (hunk 4 above)
Comment (armin-ilg) [reply scripts/install.sh]: Ah, I needed this when using my local DD4hep, so yes, it should work without

**Comment by armin-ilg** [comment]
@andresailer @BrieucF : Last comments addressed. As mentioned in the Software+Detector concept meeting, the geometry is not final yet though (some components missing), but I think it's still good to merge.

File: utils/dd4hep2root (hunk 6 above)
Comment (andresailer) [reply utils/dd4hep2root]: I don't see any `vars` at the link you posted?

**Comment by BrieucF** [comment]
Hi Armin, great thanks!

Actually there is conflict with https://github.com/key4hep/k4geo/pull/282 since both are putting thing in IDEA_o1_v01. I would propose to rename the version introduced here as IDEA_o1_v02 because https://github.com/key4hep/k4geo/pull/282 migrates the IDEA detector as it was in FCCDetectors (and it makes sense to call this version o1_v01) while you are using a modified vertex detector + modified beampipe. What do you think?

When this PR will be merged (I hope soon), I will start adding to `IDEA_o1_v02` the detailed drift chamber and the new muon chambers!

File: lcgeoTests/CMakeLists.txt [hunk 7]
```diff
@@ -69,16 +69,18 @@ ADD_TEST( t_${test_name} "${CMAKE_INSTALL_PREFIX}/bin/run_test_${PackageName}.sh
...
   ddsim --steeringFile=${CMAKE_CURRENT_SOURCE_DIR}/../example/steeringFile.py --compactFile=${CMAKE_CURRENT_SOURCE_DIR}/../CLIC/compact/CLIC_o2_v04/CLIC_o2_v04.xml --runType=batch -G -N=1 --outputFile=testCLIC_o2_v04.slcio )
 SET_TESTS_PROPERTIES( t_${test_name} PROPERTIES FAIL_REGULAR_EXPRESSION  "Exception;EXCEPTION;ERROR;Error" )
 
+#--------------------------------------------------
+# test for IDEA o1 v01
+SET( test_name "test_IDEA_o1_v01" )
+ADD_TEST( t_${test_name} "${CMAKE_INSTALL_PREFIX}/bin/run_test_${PackageName}.sh"
+    ddsim --compactFile=${CMAKE_CURRENT_SOURCE_DIR}/../FCCee/IDEA/compact/IDEA_o1_v01/FCCee_IDEA_o1_v01.xml --runType=batch -G -N=1 --outputFile=testIDEA_o1_v01.slcio )
+SET_TESTS_PROPERTIES( t_${test_name} PROPERTIES FAIL_REGULAR_EXPRESSION  "Exception;EXCEPTION;ERROR;Error")
+SET_TESTS_PROPERTIES( t_${test_name} PROPERTIES WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/../FCCee/IDEA/compact/IDEA_o1_v01/)
```
Comment (andresailer) [inline lcgeoTests/CMakeLists.txt:83]: Why does this have to run in the folder of the detector?

File: utils/plotstyle.py
[the remark this answers is not in the document]
Comment (armin-ilg) [reply utils/plotstyle.py:1]: Ok, sadly $FCCSW on cvmfs does not contain the scripts folder where [plotstyle.py](https://github.com/HEP-FCC/FCCSW/blob/master/Examples/scripts/plotstyle.py) lies, so I couldn't import it nicely. let me know if you have a better solution

File: scripts/geoDisplay.sh
[the remark this answers is not in the document]
Comment (armin-ilg) [reply scripts/geoDisplay.sh:1]: Right, deleted

File: scripts/build_view.sh
[the remark this answers is not in the document]
Comment (armin-ilg) [reply scripts/build_view.sh:1]: I think it's quite useful for quick checks of the geometry, many don't know of this possibility and therefore I think it's worth it to keep it here. Renamed it, let me know if you have another name in mind or still want to drop it.

File: FCCee/IDEA/compact/IDEA_o1_v01/Beampipe_o1_v01.xml
[the remark this answers is not in the document]
Comment (armin-ilg) [reply FCCee/IDEA/compact/IDEA_o1_v01/Beampipe_o1_v01.xml:1]: Done that now

File: utils/plotstyle.py
[the remark this answers is not in the document]
Comment (andresailer) [reply utils/plotstyle.py:1]: Open a ticket in FCCSW so that this gets installed?

File: FCCee/IDEA/compact/IDEA_o1_v01/Vertex_IDEA_o1_v01.xml [hunk 8]
```diff
@@ -0,0 +1,580 @@
...
+            <argument value="v_y=1."/>
+            <argument value="n_z=-1."/>
+        </plugin>
+    </plugins>
+
+    <!-- Vertex inner barrel support structure imported via DDCAD, material budget evaluation with g4MaterialScan is working, but not with the k4SimGeant4 methods -->
+    <detectors>
+        <detector id="0" name="VTXIBSupport" type="DD4hep_TestShape_Creator">
+            <check>
+                <shape type="CAD_MultiVolume" ref="${K4GEO}/share/k4geo/FCCee/IDEA/compact/IDEA_o1_v01/VertexSupport_o1_v01.obj" unit="mm"/>
```
Comment (andresailer) [inline FCCee/IDEA/compact/IDEA_o1_v01/Vertex_IDEA_o1_v01.xml:574]: You are saying, you cannot use a relative path here?

File: FCCee/IDEA/compact/IDEA_o1_v01/Beampipe_o1_v01.xml
Comment (andresailer) [inline FCCee/IDEA/compact/IDEA_o1_v01/Beampipe_o1_v01.xml:1]: Please use the same name as the source [Beampipe_o4_v05.xml](https://github.com/key4hep/k4geo/blob/master/FCCee/CLD/compact/CLD_o2_v05/Beampipe_o4_v05.xml)

File: utils/plotstyle.py
[the remark this answers is not in the document]
Comment (armin-ilg) [reply utils/plotstyle.py:1]: Okay, done: https://github.com/HEP-FCC/FCCSW/issues/462

File: FCCee/IDEA/compact/IDEA_o1_v01/Vertex_IDEA_o1_v01.xml (hunk 8 above)
Comment (armin-ilg) [reply FCCee/IDEA/compact/IDEA_o1_v01/Vertex_IDEA_o1_v01.xml:574]: Yes, just using ref="VertexSupport_o1_v01.obj" will only work if it's run inside of the FCCee/IDEA/compact/IDEA_o1_v01 folder. I can open a ticket in DD4hep for that if you think this should be fixed (would be useful I think)

File: FCCee/IDEA/compact/IDEA_o1_v01/Beampipe_o1_v01.xml
Comment (armin-ilg) [reply FCCee/IDEA/compact/IDEA_o1_v01/Beampipe_o1_v01.xml:1]: Yes, you're right, as long as they are identical, they should have the same name, thanks! Changed!

File: FCCee/IDEA/compact/IDEA_o1_v01/Vertex_IDEA_o1_v01.xml (hunk 8 above)
Comment (andresailer) [reply FCCee/IDEA/compact/IDEA_o1_v01/Vertex_IDEA_o1_v01.xml:574]: Yes please open an issue. We must not depend on running in the same folder or having the full path.

File: lcgeoTests/CMakeLists.txt [hunk 9]
```diff
@@ -69,6 +69,13 @@ ADD_TEST( t_${test_name} "${CMAKE_INSTALL_PREFIX}/bin/run_test_${PackageName}.sh
   ddsim --steeringFile=${CMAKE_CURRENT_SOURCE_DIR}/../example/steeringFile.py --compactFile=${CMAKE_CURRENT_SOURCE_DIR}/../CLIC/compact/CLIC_o2_v04/CLIC_o2_v04.xml --runType=batch -G -N=1 --outputFile=testCLIC_o2_v04.slcio )
 SET_TESTS_PROPERTIES( t_${test_name} PROPERTIES FAIL_REGULAR_EXPRESSION  "Exception;EXCEPTION;ERROR;Error" )
 
+#--------------------------------------------------
+# test for IDEA o1 v01
+SET( test_name "test_IDEA_o1_v01" )
+ADD_TEST( t_${test_name} "${CMAKE_INSTALL_PREFIX}/bin/run_test_${PackageName}.sh"
+    ddsim --compactFile=${CMAKE_CURRENT_SOURCE_DIR}/../FCCee/IDEA/compact/IDEA_o1_v01/FCCee_IDEA_o1_v01.xml --runType=batch -G -N=1 --outputFile=testIDEA_o1_v01.slcio )
+SET_TESTS_PROPERTIES( t_${test_name} PROPERTIES FAIL_REGULAR_EXPRESSION  "Exception;EXCEPTION;ERROR;Error")
+SET_TESTS_PROPERTIES( t_${test_name} PROPERTIES)
```
Comment (andresailer) [inline lcgeoTests/CMakeLists.txt:78]: This line is breaking cmake

Comment (armin-ilg) [reply lcgeoTests/CMakeLists.txt]: Changed, sorry!

File: FCCee/IDEA/compact/IDEA_o1_v01/Vertex_IDEA_o1_v01.xml (hunk 8 above)
Comment (armin-ilg) [reply FCCee/IDEA/compact/IDEA_o1_v01/Vertex_IDEA_o1_v01.xml:574]: Done: https://github.com/AIDASoft/DD4hep/issues/1166

**Comment by andresailer** [comment]
This is going to conflict with #282 for the IDEA_o1_v01

**Comment by BrieucF** [comment]
> Actually there is conflict with #282 since both are putting thing in IDEA_o1_v01. I would propose to rename the version introduced here as IDEA_o1_v02 because #282 migrates the IDEA detector as it was in FCCDetectors (and it makes sense to call this version o1_v01) while you are using a modified vertex detector + modified beampipe. What do you think?
> 
> When this PR will be merged (I hope soon), I will start adding to `IDEA_o1_v02` the detailed drift chamber and the new muon chambers!

**Comment by armin-ilg** [comment]
Hi. So I have done the last additions to the IDEA vertex. It's now also possible to do some very simple (and by far not final) digitisation using k4RecTracker (my fork is [here](https://github.com/armin-ilg/k4RecTracker)) without needing to use the k4MarlinWrapper.

The IDEA vertex detector is now quite accurately described. In comparison to the MDI note, also the end-of-stave/ladder structures have been implemented. Only the large holder structure for the disks is missing.

Anything else that needs to be changed @andresailer ?

**Comment by andresailer** [comment]
What Brieuc posted above again.

**Comment by andresailer** [comment]
Also check yourself why the tests are failing please.

**Comment by armin-ilg** [comment]
> Also check yourself why the tests are failing please.

Done. I commented the vertex inner barrel support for the moment, as this would prevent the test from passing. 
(see previous problem)
> Yes, just using ref="VertexSupport_o1_v01.obj" will only work if it's run inside of the FCCee/IDEA/compact/IDEA_o1_v01 folder. I can open a ticket in DD4hep for that if you think this should be fixed (would be useful I think)

**Comment by armin-ilg** [comment]
> What Brieuc posted above again.

@andresailer : So if I understand Brieuc's comment correctly
> Actually there is conflict with https://github.com/key4hep/k4geo/pull/282 since both are putting thing in IDEA_o1_v01. I would propose to rename the version introduced here as IDEA_o1_v02 because https://github.com/key4hep/k4geo/pull/282 migrates the IDEA detector as it was in FCCDetectors (and it makes sense to call this version o1_v01) while you are using a modified vertex detector + modified beampipe. What do you think?
> When this PR will be merged (I hope soon), I will start adding to IDEA_o1_v02 the detailed drift chamber and the new muon chambers!

this means that we can merge this PR now and then the other open PR will build on this IDEA version, right? So we can go ahead and merge this PR from your side @BrieucF ?

**Comment by BrieucF** [comment]
> > What Brieuc posted above again.
> 
> @andresailer : So if I understand Brieuc's comment correctly
> 
> > Actually there is conflict with #282 since both are putting thing in IDEA_o1_v01. I would propose to rename the version introduced here as IDEA_o1_v02 because #282 migrates the IDEA detector as it was in FCCDetectors (and it makes sense to call this version o1_v01) while you are using a modified vertex detector + modified beampipe. What do you think?
> > When this PR will be merged (I hope soon), I will start adding to IDEA_o1_v02 the detailed drift chamber and the new muon chambers!
> 
> this means that we can merge this PR now and then the other open PR will build on this IDEA version, right? So we can go ahead and merge this PR from your side @BrieucF ?

Hi Armin, sorry I was not clear. No, we should first rename the folder `IDEA_o1_v01` --> `IDEA_o1_v02`, rename `IDEA_o1_v01.xml` --> `IDEA_o1_v02.xml` and modify the relevant paths in affected scripts. Then it is good for merging on my side.

**Comment by armin-ilg** [comment]
> > > What Brieuc posted above again.
> > 
> > 
> > @andresailer : So if I understand Brieuc's comment correctly
> > > Actually there is conflict with #282 since both are putting thing in IDEA_o1_v01. I would propose to rename the version introduced here as IDEA_o1_v02 because #282 migrates the IDEA detector as it was in FCCDetectors (and it makes sense to call this version o1_v01) while you are using a modified vertex detector + modified beampipe. What do you think?
> > > When this PR will be merged (I hope soon), I will start adding to IDEA_o1_v02 the detailed drift chamber and the new muon chambers!
> > 
> > 
> > this means that we can merge this PR now and then the other open PR will build on this IDEA version, right? So we can go ahead and merge this PR from your side @BrieucF ?
> 
> Hi Armin, sorry I was not clear. No, we should first rename the folder `IDEA_o1_v01` --> `IDEA_o1_v02`, rename `IDEA_o1_v01.xml` --> `IDEA_o1_v02.xml` and modify the relevant paths in affected scripts. Then it is good for merging on my side.

Ah, sorry for the confusion. Changed now!
(the Silicon wrapper will be added in a future PR, although I have a preliminary version already)

**Pull request merged**